src/PlaywrightSharp.Abstractions/IResponse.cs

```diff
@@ -49,5 +50,17 @@ public interface IResponse
         /// </summary>
         /// <returns>A <see cref="Task"/> that completes when the text was processed, yielding to a text representation of response body.</returns>
         Task<string> GetTextAsync();
+
+        /// <summary>
+        /// Returns a <see cref="Task"/> which resolves to a <see cref="JsonDocument"/> representation of response body.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> that completes when the json body is parsed, yielding a <see cref="JsonDocument"/> representation of response body.</returns>
+        Task<JsonDocument> GetJsonAsync();
+
+        /// <summary>
+        /// Returns a <see cref="Task"/> which resolves to a buffer with response body.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> that completes when the response is returned by the server, yielding a <see cref="byte"/> array.</returns>
+        Task<byte[]> GetBufferAsync();

```

Comment:
can this be `Span<byte>`?

Reply:
What would be the benefit for the consumer?

Reply:
I just checked that you can implicitly cast T[] to Span<T> ([source](https://docs.microsoft.com/en-us/archive/msdn-magazine/2018/january/csharp-all-about-span-exploring-a-new-net-mainstay).

I think a byte[] as a more pure and generic type.